starter-code/src/viking.js

```diff
@@ -1,11 +1,75 @@
 // Soldier
-function Soldier() {}
+function Soldier (healthArg, strengthArg) {
+    this.health = healthArg;
+    this.strength = strengthArg;
+
+}
+
 
 // Viking
-function Viking() {}
+function Viking(name,healthArg, strengthArg) {
+    Soldier.call(this, healthArg, strengthArg);
+    this.name = name;
+}
 
 // Saxon
-function Saxon() {}
+function Saxon(healthArg, strengthArg) {
+    Soldier.call(this, healthArg, strengthArg);
+
+}
 
 // War
-function War() {}
+function War() {
+    this.vikingArmy = [];
+    this.saxonArmy = [];
+}
+
+War.prototype.addViking = function (Viking){
+    this.vikingArmy.push(Viking);
+}
+War.prototype.addSaxon = function (Saxon){
+    this.saxonArmy.push(Saxon);
+}
+War.prototype.vikingAttack = function (){
+    var randomSaxon = Math.floor(Math.random()*this.saxonArmy.length);
+    var force = Viking.attack();
```

Comment:
Por ejemplo en esta linea, en lugar de Viking.attack() deberia ser this.vikingArmy[0].attack()

Uno es la clase, la segunda es la instancia.